integration-tests/tests/src/tests/array-buffer.ts

```diff
@@ -0,0 +1,230 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { Buffer } from "buffer";
+import { expect } from "chai";
+import Realm from "realm";
+import { openRealmBeforeEach } from "../hooks";
+
+const isNodeProcess = typeof process === "object" && process + "" === "[object process]";
```

Comment:
I have squigglys here telling me to do the following.
![image](https://user-images.githubusercontent.com/44238293/212276314-f0515b99-d430-45b1-91ea-90a92cfcba53.png)
As I'm very unfamiliar with this area I thought it'd be worth asking here whether I should do it or not

Reply:
Nope. If you have Node.js specific tests, you can do one of three things:
- Determine if they're really Node.js specific or could be generalised a bit to make them run on every platform.
- If they really do test Node.js specific behaviour:
  - You can use the `describe.skipIf` or `it.skipIf` to skip the test if the `environment.node` isn't true. Here's [an example where we skip a test on Android](https://github.com/realm/realm-js/blob/master/integration-tests/tests/src/tests/realm-constructor.ts#L308).
  - If your test depends on importing a Node.js specific API (other than `path` or `fs` which are already injected as globals), you can move the test to the Node.js specific part of our test suite: https://github.com/realm/realm-js/tree/master/integration-tests/tests/src/node ... there you'll have all the Node.js runtime APIs available (such as the `process` global).

In this specific case, I'd probably use `it.skipIf(!environment.node, ...` for the "supports arrayBuffer-nodeBuffer" test and try removing the early return from the "supports inserting empty arrayBuffer" test to see if / why it doesn't work on other platforms as well.